src/main/java/com/github/elenterius/biomancy/init/AcidInteractions.java

```diff
@@ -157,4 +165,55 @@ else if (livingEntity.tickCount % 10 == 0 && livingEntity.getRandom().nextFloat(
 		}
 	}
 
+	private static final String TIMER_KEY = "biomancy:digestion_timer";
+	private static final float EFFICIENCY = 0.8f;
+	public static void tryDigest(ItemEntity itemEntity) {
```

Comment:
Don't call this method each tick due to the recipe lookup and the possibility of having many item entities in the world inside acid. I would prefer every 10-20 ticks.

Furthermore, cache the found recipe id and use that one first to test if the recipe is still valid.